### DISCUSSION TITLE
Mod incompatibility problems & Bugs

### DISCUSSION THREAD

**Discussion by Cliff511** [BeansGalaxy/Beans-Backpacks-2#110] [Feedback]
Hello, Me and my friends played with this mod for a while with our medieval modpack that we created. We found some problems that would be awesome if you could fix. Love the mod so I'd rather these problems be solved than changing to another backpack mod since this one is clearly just be best alternative for Fabric. I should also note that I am playing on the 1.20.1 version of minecraft Fabric.

**Incompatability with "Forgotten Graves":**

When you die with the two mods installed you will leave a grave on the ground, and when retrieving the grave you get all your items back but all the items inside the backpack specifically gets deleted. I have tried to look into the problem and I think it is because the gravestone mod can't find any of the items inside the backpack so they are not kept. I tried this with other gravestone mods and the same thing happened no matter what config I tried.

**Incompatability with "REI":**

Recipe for Netherite backpack does not show. I am not sure if this applies to JEI though, but I would guess that would be the case since the mods work in very similar ways.

**Comment by BeansGalaxy** [comment]
I'm glad you're liking the mod and I wish I could help you out but these have been long standing issues that have no easy solution.

### Graves

Grave mods and I are both fighting over the player's inventory on death. I'm trying to collect the item in the back slot before they empty the inventory but they don't give me much room to do what I need to do before they clear it. I've done what I can in terms of compatibility and there's not much else I can do.

### REI

REI/JEI/EMI all don't have a good way to implement custom smithing recipes. If you know anything about data-packs, my mod doesn't use the vanilla smithing recipe. Instead it uses one I've written myself so that I can craft items with nbt. You can register custom recipes for crafting tables in all those mods easily or register custom crafting blocks but custom recipes for vanilla crafting blocks like the loom or smithing table *to my understanding* cannot be done.

**Reply by Cliff511** [reply]
With the grave mod, Is it possible for you to make it so the backpack drops on the ground before the grave mod deletes the items?

**Reply by BeansGalaxy** [reply]
No, and that's the issue. Most grave mods try to empty the inventory as early as possible to avoid anything happening to your items. I store my items in a very unique way to get the player's backpack inventory to be ready at all times to be placed, accessed through the inventory, and viewed by another player by clicking on your back.

Since my way of storing items is unique, grave mod's need to either write specific code to handle my items before they erase the inventory or identify and ignore my inventory to let my code handle it. I know "You're in Grave Danger" & "Universal Graves" works with my mod but compatibilities with grave mods is a coin flip.
